datafusion/core/src/datasource/listing/mod.rs

```diff
@@ -32,15 +32,26 @@ pub use table::{ListingOptions, ListingTable, ListingTableConfig};
 pub type PartitionedFileStream =
     Pin<Box<dyn Stream<Item = Result<PartitionedFile>> + Send + Sync + 'static>>;
 
+/// File part identified by [start, end) positions.
```

Comment:
Are these positions measured in `bytes` from the start of the file?

Reply:
Yes, get the row group midpoint in bytes in https://github.com/apache/arrow-rs/blob/master/parquet/src/file/serialized_reader.rs#L222-L232, and check if the midpoint is in the [start, end) range.

Reply:
```suggestion
/// Only scan a subset of Row Groups from the Parquet file whose data "midpoint"
/// lies within the [start, end) byte offsets. This option can be used to scan non-overlapping
/// sections of a Parquet file in parallel. 
```